### PR TITLE
XmlSerializer with unobtrusive messages should propagate properties of message

### PR DESCRIPTION
**Pull request by danielmarbach**
Closes https://github.com/Particular/NServiceBus/issues/3975

### CODE REVIEW

File: src/NServiceBus.Core/Serializers/XML/XmlDeserialization.cs [hunk 1]
```diff
@@ -629,17 +629,21 @@ object GetPropertyValue(Type type, XmlNode n)
 
         PropertyInfo GetProperty(Type t, string name)
         {
-            IEnumerable<PropertyInfo> props;
-            cache.typeToProperties.TryGetValue(t, out props);
+            IEnumerable<PropertyInfo> properties;
+            if (!cache.typeToProperties.TryGetValue(t, out properties))
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Serializers/XML/XmlDeserialization.cs]: This is the relevant change right?

Comment (danielmarbach) [inline src/NServiceBus.Core/Serializers/XML/XmlDeserialization.cs]: Fair enough I split it out

Comment (danielmarbach) [inline src/NServiceBus.Core/Serializers/XML/XmlDeserialization.cs]: Just realized that the only formatting change was https://github.com/Particular/NServiceBus/pull/4095/commits/91d44659b7956727f4288c07852501ff126e0612#diff-36144d5d3104efccfa5e06f871ca3cd8R280
Splitting that out into dedicated commit feels like overkill

Comment (andreasohlund) [inline src/NServiceBus.Core/Serializers/XML/XmlDeserialization.cs]: ah, saw now that the other changes where in some tests. I can live with that :+1:

**Comment by danielmarbach** [comment]
@Particular/nservicebus-maintainers please review

**Comment by andreasohlund** [comment]
LGTM ( @danielmarbach walked me through it on a call)

**Pull request merged**